test/data/roundtrip/fle2aead-generated.json

```diff
@@ -0,0 +1,329 @@
+    [1]=>
+    string(41) "generated test. AD length=20, M length=64"
+    [2]=>
+    string(32) "0dea6351b7c5f722135cc71674fe2aa5"
+    [3]=>
+    string(40) "56436a1a19ecd83b1cfac30c385d2b10f09912b6"
+    [4]=>
+    string(192) "97fd8fa72107825842f7b8b6446bc15295bfddc81b24775d61d637ee79edab76bb2ea1da325427d3b1841598bfb423ce5e9cb9e2d19a5ab863971ce7df36c3dc0000000000000000000000000000000000000000000000000000000000000000"
+    [5]=>
+    string(128) "759e97fabfc00a4097051ce11b1d0303fd92f32a0c91d24048b8b3e3cc1272c07361c3f2c58af8c4e05fc7655f883f10f85c88cc3d061a7e48829923612602b6"
+    [6]=>
+    string(224) "0dea6351b7c5f722135cc71674fe2aa5a8cec9ba7111214284703319deaf50561f7868b76cc33f471cd47902ef7b7e4b00d852d16456c040b30cbf9aae510cd93db432f9304b27fdd0f9fbf361072e7a2680f55fd143bce06d67ac84851d2648550e98a297bed4087707e058db6e79ab"
+  }
+  [11]=>
+  array(7) {
+    [0]=>
+    string(927) "{     .testname = "generated test. AD length=20, M length=100",     .iv = "91ce2657c5c51b63e3f8698962986783",     .associated_data = "0bb9641dbd11aae69d90451ffa3dc785ea325c86",     .key = "7efbb28148595b9ab5fdda8b94c3f3ed5094de75396a8a7bed234cc83de62f4558601d183b1bb383588667dddb921d8c08cac48c7d7671fdb53b7d340a5c90960000000000000000000000000000000000000000000000000000000000000000",     .plaintext = "db6faefc3a0ee1d0dccc810ec663c82889370ec9ff430063defd2ba8611b870fdf1f8e0f51323f101166ac1dc4fb7db1819b5ebd16934ec18827582b74f96e38ba0caa8f616051fee488f68015870b0fed27aede025c8be1957237e1c71375d64055dcf1",     .ciphertext = "91ce2657c5c51b63e3f869896298678332d4b0ba0ef78ed3b73e4b4a507cf0c16fa6c98716a500ac75d99ff1e11a9ece604a5aab38f201c0772ac3bedb8abcf564f6cea8f65d3710ee4ca34887a5170296fee8059f852898aa9e24664940a1cd44b50881e6eb912e352ecd708850cb8345890c6eceed6b4a3200fc71f2112f7ea88ee61b9de21d3e35b4ed761f8130ca56f4e9e6",     "
+    [1]=>
+    string(42) "generated test. AD length=20, M length=100"
+    [2]=>
+    string(32) "91ce2657c5c51b63e3f8698962986783"
+    [3]=>
+    string(40) "0bb9641dbd11aae69d90451ffa3dc785ea325c86"
+    [4]=>
+    string(192) "7efbb28148595b9ab5fdda8b94c3f3ed5094de75396a8a7bed234cc83de62f4558601d183b1bb383588667dddb921d8c08cac48c7d7671fdb53b7d340a5c90960000000000000000000000000000000000000000000000000000000000000000"
+    [5]=>
+    string(200) "db6faefc3a0ee1d0dccc810ec663c82889370ec9ff430063defd2ba8611b870fdf1f8e0f51323f101166ac1dc4fb7db1819b5ebd16934ec18827582b74f96e38ba0caa8f616051fee488f68015870b0fed27aede025c8be1957237e1c71375d64055dcf1"
+    [6]=>
+    string(296) "91ce2657c5c51b63e3f869896298678332d4b0ba0ef78ed3b73e4b4a507cf0c16fa6c98716a500ac75d99ff1e11a9ece604a5aab38f201c0772ac3bedb8abcf564f6cea8f65d3710ee4ca34887a5170296fee8059f852898aa9e24664940a1cd44b50881e6eb912e352ecd708850cb8345890c6eceed6b4a3200fc71f2112f7ea88ee61b9de21d3e35b4ed761f8130ca56f4e9e6"
+  }
+}
```

Comment:
Woah. Hrmmm good news: This caught a failure where malformed JSON is incorrectly returned as an empty object. Looking into that now.

Reply:
Nice! I think we'll have to fix [this line](https://github.com/mongodb/libmongocrypt/blob/f02149ad61e7bdd6bfb02cbaee0aee408160138e/test/test-mongocrypt.c#L63), because the prior call to `bson_json_reader_read` returns `-1` for malformed JSON, instead of 0, hence it returned an empty BSON instead of aborting.

Reply:
Gah. Couldn't see the forest through the trees, that's twice this week you've blown my cases wide open.